src/sentry/models/sentryapp.py

```diff
@@ -45,10 +46,32 @@ class Meta:
         app_label = 'sentry'
         db_table = 'sentry_sentryapp'
 
+    @property
+    def organizations(self):
+        if not self.pk:
+            return Organization.objects.none()
+
+        return Organization \
+            .objects \
+            .select_related('sentry_app_installations') \
+            .filter(sentry_app_installations__sentry_app_id=self.id)
+
+    @property
+    def teams(self):
+        from sentry.models import Team
+
+        if not self.pk:
+            return Team.objects.none()
+
+        return Team.objects.filter(organization__in=self.organizations)
```

Comment:
Should this be memoized at the instance level? Accessing a property multiple times per-request probably shouldn't emit several queries.

Reply:
I'm not aware of a good way to deal with the before/after save boundary. This can be revisited if it becomes a problem.